lib/data/external/GCP/GcpApis/completeMPU.js

```diff
@@ -0,0 +1,232 @@
+        }
+    }
+
+    const _copyToOverflow = (numParts, callback) => {
+        // copy phase: in overflow bucket
+        // resetting component count by moving item between
+        // different region/class buckets
+        logger.trace('completeMultipartUpload: copy to overflow',
+            { partCount: numParts });
+        const parts = createMpuList(params, 'mpu2', numParts);
+        let doneCount = 0;
+        if (parts.length !== numParts) throw errors.InternalError;
+        return async.each(parts, (infoParts, cb) => {
+            const partName = infoParts.PartName;
+            const partNumber = infoParts.PartNumber;
+            const overflowKey = createMpuKey(
+                params.Key, params.UploadId, partNumber, 'overflow');
+            const rewriteParams = {
+                SourceBucket: params.MPU,
+                SourceObject: partName,
+                DestinationBucket: params.Overflow,
+                DestinationObject: overflowKey,
+            };
+            logger.trace('rewrite object', { rewriteParams });
+            let rewriteDone = false;
+            async.whilst(() => !rewriteDone, moveOn => {
+                this.rewriteObject(rewriteParams, (err, res) => {
+                    if (err) {
+                        logHelper(logger, 'error', 'err in ' +
+                            'createMultipartUpload - rewriteObject', err);
+                    } else {
+                        rewriteDone = res.done;
+                        rewriteParams.RewriteToken = res.rewriteToken;
+                    }
+                    return moveOn(err);
+                });
+            }, err => {
+                if (!err) {
+                    doneCount++;
+                }
+                return cb(err);
+            });
+        }, err => {
+            if (err) {
+                return callback(err);
+            }
+            return callback(null, doneCount);
+        });
+    };
+
+    const _composeOverflow = (numParts, callback) => {
+        // final compose: in overflow bucket
+        // number of parts to compose <= 10
+        // perform final compose in overflow bucket
+        logger.trace('completeMultipartUpload: overflow compose');
+        const parts = createMpuList(params, 'overflow', numParts);
+        const partList = parts.map(item => (
+            { PartName: item.PartName }));
+        if (partList.length < 2) {
+            logger.trace(
+                'fewer than 2 parts in overflow, skip to copy phase');
+            return callback(null, partList[0].PartName);
+        }
+        const composeParams = {
+            Bucket: params.Overflow,
+            Key: createMpuKey(params.Key, params.UploadId, 'final'),
+            MultipartUpload: { Parts: partList },
+        };
+        return _retryCompose.call(this, composeParams, 0, err => {
+            if (err) {
+                return callback(err);
+            }
+            return callback(null, null);
```

Comment:
Unless there's a good reason to provide `null` values, we usually just call `callback()`